1.segment_images/scripts/2.segment_whole_organoids.py

```diff
@@ -0,0 +1,243 @@
+
+
+# In[4]:
+
+
+# find the cytoplasmic channels in the image set
+for f in files:
+    if "405" in f:
+        nuclei = io.imread(f)
+    elif "488" in f:
+        cyto1 = io.imread(f)
+    elif "555" in f:
+        cyto2 = io.imread(f)
+    elif "640" in f:
+        cyto3 = io.imread(f)
+    elif "TRANS" in f:
+        brightfield = io.imread(f)
+    else:
+        print(f"Unknown channel: {f}")
+
+cyto = np.max([cyto1, cyto2, cyto3], axis=0)
+# pick which channels to use for cellpose
+cyto = skimage.exposure.equalize_adapthist(cyto1, clip_limit=clip_limit)
+
+
+original_cyto_image = cyto.copy()
+
+original_cyto_z_count = cyto.shape[0]
+
+
+# In[5]:
+
+
+# make a 2.5 D max projection image stack with a sliding window of 3 slices
+image_stack_2_5D = np.empty((0, cyto.shape[1], cyto.shape[2]), dtype=cyto.dtype)
+for image_index in range(cyto.shape[0]):
+    image_stack_window = cyto[image_index : image_index + window_size]
+    if not image_stack_window.shape[0] == window_size:
+        break
+    # guassian blur the image stack
+    image_stack_window = skimage.filters.gaussian(image_stack_window, sigma=1)
+    # max project the image stack
+    image_stack_2_5D = np.append(
+        image_stack_2_5D, np.max(image_stack_window, axis=0)[np.newaxis, :, :], axis=0
+    )
+
+image_stack_2_5D = np.array(image_stack_2_5D)
+cyto = np.array(image_stack_2_5D)
+print("2.5D cyto image stack shape:", cyto.shape)
+
+
+# Use butterworth FFT filter to remove high frequency noise :)
+imgs = skimage.filters.butterworth(
+    cyto,
+    cutoff_frequency_ratio=0.1,
+    high_pass=False,
+    order=5.0,
+    squared_butterworth=True,
+)
+if in_notebook:
+    # plot the nuclei and the cyto channels
+    plt.figure(figsize=(10, 10))
+    plt.subplot(121)
+    plt.imshow(cyto[9, :, :], cmap="gray")
+    plt.title("Raw cyto")
+    plt.axis("off")
+    plt.subplot(122)
+    plt.imshow(imgs[9, :, :], cmap="gray")
+    plt.title(f"Butterworth filtered cyto")
+    plt.axis("off")
+    plt.show()
+
+
+# In[6]:
+
+
+# if in_notebook:
+#     # plot the nuclei and the cyto channels
+#     plt.figure(figsize=(10, 10))
+#     plt.subplot(121)
+#     plt.imshow(cyto[9, :, :], cmap="gray")
+#     plt.title("Raw cyto")
+#     plt.axis("off")
+#     plt.subplot(122)
+#     plt.imshow(imgs[9, :, :], cmap="gray")
+#     plt.title(f"Gaussian blur: sigma {sigma}")
+#     plt.axis("off")
+#     plt.show()
+
+
+# In[7]:
+
+
+use_GPU = torch.cuda.is_available()
+# Load the model
+model_name = "cyto3"
+model = models.CellposeModel(gpu=use_GPU, model_type=model_name)
+
+# Perform segmentation
+labels, details, _ = model.eval(
+    imgs,
+    channels=[0, 0],
+    z_axis=0,
+    stitch_threshold=0.8,
+    diameter=750,
+)
+
+
+# In[8]:
+
+
+# reverse sliding window max projection
+full_mask_z_stack = []
+reconstruction_dict = {index: [] for index in range(original_cyto_z_count)}
+print(f"Decoupling the sliding window max projection of {window_size} slices")
```

Comment:
Consider improving the documentation by defining what the reverse sliding window max projection does and also why these two variables are important, what are you storing and how is it going to be used (you can explain that below the code).